src/go/rpk/pkg/cli/transform/init.go

```diff
@@ -0,0 +1,257 @@
+/*
+* Copyright 2023 Redpanda Data, Inc.
+*
+* Use of this software is governed by the Business Source License
+* included in the file licenses/BSL.md
+*
+* As of the Change Date specified in that file, in accordance with
+* the Business Source License, use of this software will be governed
+* by the Apache License, Version 2.0
+ */
+
+package transform
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/buildpack"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/project"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cli/transform/template"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type optionalBool struct {
+	ok *bool
+}
+
+func (b *optionalBool) Set(s string) error {
+	if strings.ToLower(s) == "prompt" {
+		b.ok = nil
+		return nil
+	}
+	v, err := strconv.ParseBool(s)
+	b.ok = &v
+	return err
+}
+
+func (*optionalBool) Type() string {
+	return "bool"
+}
+
+func (b *optionalBool) String() string {
+	if b.ok == nil {
+		return "prompt"
+	}
+	return strconv.FormatBool(*b.ok)
+}
+
+func (*optionalBool) IsBoolFlag() bool { return true }
+
+func newInitializeCommand(fs afero.Fs) *cobra.Command {
+	var (
+		name    string
+		lang    project.WasmLang
+		install optionalBool
+	)
+	cmd := &cobra.Command{
+		Use:   "init [DIRECTORY]",
+		Short: "Initialize a transform",
+		Long: `Initialize a transform.
+
+Creates a new transform using a template in the current directory.
+
+A new directory can be created by specifying it in the command, like:
+
+rpk transform init foobar
```

Comment:
[nit] can you add two spaces before the command we want to use as an example?

```suggestion
  rpk transform init foobar
```
we have tried this in other commands: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/registry/schema/create.go#L49

Reply:
Done! Thanks :)